src/custom_request_schedule_manager.cc

```diff
@@ -118,14 +100,35 @@ CustomRequestScheduleManager::GetScheduleFromDataset() const
 
   const size_t dataset_size{data_loader_->GetTotalSteps(0)};
 
-  for (size_t dataset_index{0}; dataset_index < dataset_size; ++dataset_index) {
+  if (dataset_size == 0) {
+    throw std::runtime_error(
+        "Expected input data JSON to have at least one request payload.");
+  }
+
+  if (warmup_request_count_ >= dataset_size) {
+    throw std::runtime_error(
+        "Expected warmup request count to be less than the dataset size.");
+  }
+
+  if (dataset_size != warmup_request_count_ + benchmark_request_count_) {
+    throw std::runtime_error(
+        "Expected input data JSON size to be equal to the warmup request count "
+        "plus the benchmark request count.");
+  }
```

Comment:
Just extra validation.

Some of this was already checked in command_line_parser.cc, but it's better to re-check it here in case developers change expectations before this code in future.

Reply:
I mean, the check is a bit redundant with the one just above? The one on line 108 is checking if `dataset_size` is bigger or equal to `warmup_request_count_`, but then on 113, you ensure it's strictly equal. Not really a big deal, just maybe a tiny bit weird.

Reply:
Yeah, you're spot on.

Removed this check and the `benchmark_request_count_` data member.